manager/src/main/java/com/redhat/service/bridge/manager/BridgeWorker.java

```diff
@@ -0,0 +1,208 @@
+        preparingWorkerDAO.deleteById(preparingWorker.getId());
+
+        // The bridge is ready to be provisioned by the shard.
+        bridge.setStatus(ManagedEntityStatus.DEPROVISION);
+        bridge.setDesiredStatus(ManagedEntityStatus.DELETING);
+        transact(bridge);
+    }
+
+    @Override
+    @Scheduled(every = "30s", concurrentExecution = Scheduled.ConcurrentExecution.SKIP)
+    public void reschedule() {
+        LOGGER.info("rescheduled");
+        List<PreparingWorker> preparingWorkers = preparingWorkerDAO.findByWorkerIdAndStatusAndType(workerIdProvider.getWorkerId(), BridgeWorkerStatus.FAILURE.toString(), "BRIDGE");
+        for (PreparingWorker preparingWorker : preparingWorkers) {
+            Bridge bridge = bridgeDAO.findById(preparingWorker.getEntityId());
+            routePreparingWorker(preparingWorker, bridge);
+        }
+    }
+
+    /**
+     * Every 5 minutes it will retrieve all the instances that have not been updated for more than 30 minutes.
+     */
+    @Override
+    @Scheduled(every = "5m", concurrentExecution = Scheduled.ConcurrentExecution.SKIP)
+    public void discoverOrphanWorkers() {
+        LOGGER.info("discovering orphan");
+        ZonedDateTime orphans = ZonedDateTime.now(ZoneOffset.UTC).minusMinutes(30); // entities older than 30 minutes will be processed with an override of the worker_id.
+        List<PreparingWorker> preparingWorkers = preparingWorkerDAO.findByAgeAndStatusAndType(orphans, BridgeWorkerStatus.FAILURE.toString(), "BRIDGE");
+        for (PreparingWorker preparingWorker : preparingWorkers) {
+
+            // take ownership of the resource. If another instance tried to take ownership of it the transaction would fail due to the Version conflict.
+            preparingWorker.setWorkerId(workerIdProvider.getWorkerId());
+            transact(preparingWorker);
+
+            Bridge bridge = bridgeDAO.findById(preparingWorker.getEntityId());
+            routePreparingWorker(preparingWorker, bridge);
+        }
+    }
+
+    @Transactional
+    private void transact(Bridge b) {
+        bridgeDAO.getEntityManager().merge(b);
+    }
+
+    @Transactional
+    private void transact(PreparingWorker b) {
+        preparingWorkerDAO.getEntityManager().merge(b);
+    }
+
+    private PreparingWorker updatePreparingWorker(String entityId, BridgeWorkerStatus status) {
+        PreparingWorker preparingWorker = preparingWorkerDAO.findByEntityId(entityId).get(0);
+        preparingWorker.setModifiedAt(ZonedDateTime.now(ZoneOffset.UTC));
+        preparingWorker.setStatus(status.toString()); // where you are. The rescheduler will fire the event according to the desiredStatus
+        transact(preparingWorker);
+        return preparingWorker;
+    }
+
+    private void routePreparingWorker(PreparingWorker preparingWorker, Bridge bridge) {
```

Comment:
My feeling is that what the worker needs to do should be encapsulated on the Bridge entity somehow. It may be that the Worker has the ability to add metadata/status to the `Bridge` somehow to track what it is doing.

That way the worker can simply be passed a `Bridge` Entity and then work out what it needs to do from the status of the `Bridge` and making calls to the other remote services to determine if all resources have been provisioned. _How_ the worker is initiated then becomes a separate concern